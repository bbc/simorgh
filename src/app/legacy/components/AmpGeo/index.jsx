import React from 'react';

const configuration = {
  AmpBind: true,
  ISOCountryGroups: {
    eea: [
      'at',
      'ax',
      'be',
      'bg',
      'bl',
      'cy',
      'cz',
      'de',
      'dk',
      'ea',
      'ee',
      'es',
      'fi',
      'fr',
      'gb',
      'gf',
      'gg',
      'gi',
      'gp',
      'gr',
      'hr',
      'hu',
      'ic',
      'ie',
      'im',
      'is',
      'it',
      'je',
      'li',
      'lt',
      'lu',
      'lv',
      'mf',
      'mq',
      'mt',
      'nc',
      'nl',
      'no',
      'pf',
      'pl',
      'pm',
      'pt',
      're',
      'ro',
      'se',
      'si',
      'sj',
      'sk',
      'tf',
      'va',
      'wf',
      'yt',
    ],
    gbOrUnknown: ['gb', 'gg', 'im', 'je', 'uk', 'unknown'],
  },
};

export const AMP_GEO_SCRIPT = (
  <script
    async
    custom-element="amp-geo"
    src="https://cdn.ampproject.org/v0/amp-geo-0.1.js"
  />
);

const AmpGeo = () => (
  <amp-geo layout="nodisplay">
    <script
      type="application/json"
      // eslint-disable-next-line react/no-danger
      dangerouslySetInnerHTML={{ __html: JSON.stringify(configuration) }}
    />
  </amp-geo>
);

export default AmpGeo;
