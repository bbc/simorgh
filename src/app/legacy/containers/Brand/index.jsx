import React, { useContext } from 'react';
import styled from '@emotion/styled';
import Brand from '#psammead/psammead-brand/src';
import { bool, node, oneOfType, func, shape, any } from 'prop-types';
import { useTheme } from '@emotion/react';
import { ServiceContext } from '../../../contexts/ServiceContext';

const StyledBrand = styled(Brand)`
  position: relative;
  z-index: 1;
  svg {
    fill: currentColor;
    @media screen and (forced-colors: active) {
      fill: linkText;
    }
  }
`;

const BrandContainer = ({ skipLink, scriptLink, brandRef, ...props }) => {
  const { product, serviceLocalizedName, service } = useContext(ServiceContext);
  const { brandSVG } = useTheme();
  const svgMaxHeight = 24;
  const svgMinHeight = 16;
  const svgRatio = brandSVG && brandSVG.ratio;
  const minWidth = svgRatio * svgMinHeight;
  const maxWidth = svgRatio * svgMaxHeight;

  return (
    <StyledBrand
      product={product}
      serviceLocalisedName={serviceLocalizedName}
      svgHeight={svgMaxHeight}
      minWidth={minWidth}
      maxWidth={maxWidth}
      svg={brandSVG}
      url={`/${service}`}
      skipLink={skipLink}
      scriptLink={scriptLink}
      ref={brandRef}
      {...props}
    />
  );
};

BrandContainer.propTypes = {
  borderTop: bool,
  borderBottom: bool,
  skipLink: node,
  scriptLink: node,
  // eslint-disable-next-line react/forbid-prop-types
  brandRef: oneOfType([func, shape({ current: any })]),
};

BrandContainer.defaultProps = {
  borderTop: false,
  borderBottom: false,
  skipLink: null,
  scriptLink: null,
  brandRef: null,
};

export default BrandContainer;
