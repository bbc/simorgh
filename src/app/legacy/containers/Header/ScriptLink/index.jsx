import React, { useContext } from 'react';
import { compile } from 'path-to-regexp';
import clone from 'ramda/src/clone';
import { string } from 'prop-types';
import { useRouteMatch } from 'react-router-dom';
import ScriptLink from '#psammead/psammead-script-link/src';
import { UserContext } from '#contexts/UserContext';
import useToggle from '#hooks/useToggle';
import { ServiceContext } from '../../../../contexts/ServiceContext';

export const getVariantHref = ({
  path,
  params,
  service,
  variant,
  scriptSwitchId,
}) => {
  const fallback = `/${service}/${variant}`;

  // On error pages, we may not be on a path defined in router config.
  // In this case, link to the fallback.
  if (!path || path === '/') {
    return fallback;
  }

  const pathParams = clone(params);
  if (scriptSwitchId) {
    pathParams.id = scriptSwitchId;
  }

  try {
    return compile(path)(
      {
        ...pathParams,
        variant: `/${variant}`,
        amp: undefined, // we don't want to link to AMP pages directly
      },
      {
        encode: value => value,
      },
    );
  } catch {
    return fallback;
  }
};

const ScriptLinkContainer = ({ scriptSwitchId }) => {
  const { setPreferredVariantCookie } = useContext(UserContext);
  const { service, script, scriptLink } = useContext(ServiceContext);
  const { enabled: scriptLinkEnabled } = useToggle('scriptLink');
  const { enabled: variantCookieEnabled } = useToggle('variantCookie');
  const { path, params } = useRouteMatch();

  if (!scriptLinkEnabled) {
    return null;
  }

  const { text, variant } = scriptLink;

  return (
    <ScriptLink
      script={script}
      service={service}
      href={getVariantHref({
        path,
        params,
        service,
        variant,
        scriptSwitchId,
      })}
      variant={variant}
      onClick={() => {
        return (
          variantCookieEnabled && setPreferredVariantCookie(service, variant)
        );
      }}
    >
      {text}
    </ScriptLink>
  );
};

ScriptLinkContainer.propTypes = {
  scriptSwitchId: string,
};

ScriptLinkContainer.defaultProps = {
  scriptSwitchId: '',
};

export default ScriptLinkContainer;
