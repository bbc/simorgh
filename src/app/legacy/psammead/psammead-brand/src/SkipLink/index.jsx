import styled from '@emotion/styled';
import { oneOf, string } from 'prop-types';
import {
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_2_SCREEN_WIDTH_MAX,
} from '#psammead/gel-foundations/src/breakpoints';
import { GEL_SPACING } from '#psammead/gel-foundations/src/spacings';
import { getPica } from '#psammead/gel-foundations/src/typography';
import { getSansRegular } from '#psammead/psammead-styles/src/font-styles';

const SKIP_LINK_COLOR = '#333';
const SKIP_LINK_BORDER = '0.1875rem'; // 3px
const TOP_BOTTOM_SPACING = '0.75rem'; // 12px

const SkipLink = styled.a`
  position: absolute;
  clip-path: inset(100%);
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  width: 1px;
  overflow: hidden;
  padding: ${TOP_BOTTOM_SPACING} ${GEL_SPACING};
  background-color: ${props => props.theme.palette.WHITE};
  border: ${SKIP_LINK_BORDER} solid #000;
  color: ${SKIP_LINK_COLOR};
  text-decoration: none;
  ${({ script }) => script && getPica(script)};
  ${({ service }) => service && getSansRegular(service)};

  &:focus {
    clip-path: none;
    clip: auto;
    height: auto;
    width: auto;
    top: 0;
    ${({ dir }) => `
      ${dir === 'ltr' ? 'left' : 'right'}: 0;
    `}

    @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
      top: ${GEL_SPACING};
    }
  }

  @media (max-width: ${GEL_GROUP_2_SCREEN_WIDTH_MAX}) {
    padding: ${GEL_SPACING};
  }
`;

SkipLink.propTypes = {
  dir: oneOf(['ltr', 'rtl']),
  className: string,
};

SkipLink.defaultProps = { dir: 'ltr' };
SkipLink.defaultProps = { className: 'focusIndicatorRemove' };

export default SkipLink;
