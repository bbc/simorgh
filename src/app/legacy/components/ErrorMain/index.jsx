import React from 'react';
import { string, arrayOf, shape } from 'prop-types';
import styled from '@emotion/styled';
import InlineLink from '#psammead/psammead-inline-link/src';
import Paragraph from '#psammead/psammead-paragraph/src';
import { getSerifMedium } from '#psammead/psammead-styles/src/font-styles';
import {
  getCanon,
  getParagon,
  GEL_FF_REITH_SANS,
} from '#psammead/gel-foundations/src/typography';
import idSanitiser from '#lib/utilities/idSanitiser';
import Grid, { GelPageGrid } from '#components/Grid';
import { scriptPropType } from '#psammead/gel-foundations/src/prop-types';

const StatusCode = styled.span`
  ${props => (props.script ? getParagon(props.script) : '')}
  color: ${props => props.theme.palette.POSTBOX};
  display: block;
  font-family: ${GEL_FF_REITH_SANS};
  font-weight: 600;
  padding: 2.5rem 0 0.5rem 0;
`;

const Heading = styled.h1`
  ${({ script }) => script && getCanon(script)}
  ${({ service }) => getSerifMedium(service)}
  color: ${props => props.theme.palette.SHADOW};
  margin-top: 0;
`;

const StyledGelPageGrid = styled(GelPageGrid)`
  padding-bottom: 4rem;
`;

const CustomParagraph = styled(Paragraph)`
  padding-top: 0.2rem;
`;

const ErrorMain = ({
  statusCode,
  title,
  message,
  solutions,
  callToActionFirst,
  callToActionLinkText,
  callToActionLinkUrl,
  callToActionLast,
  script,
  service,
}) => (
  <StyledGelPageGrid
    as="main"
    role="main"
    columns={{
      group0: 6,
      group1: 6,
      group2: 6,
      group3: 6,
      group4: 8,
      group5: 20,
    }}
    enableGelGutters
  >
    <Grid
      item
      startOffset={{
        group0: 1,
        group1: 1,
        group2: 1,
        group3: 1,
        group4: 2,
        group5: 5,
      }}
      columns={{
        group0: 6,
        group1: 6,
        group2: 6,
        group3: 6,
        group4: 6,
        group5: 12,
      }}
      margins={{ group0: true, group1: true, group2: true, group3: true }}
    >
      <StatusCode script={script} data-e2e="status-code">
        {statusCode}
      </StatusCode>
      <Heading id="content" script={script} service={service} tabIndex="-1">
        {title}
      </Heading>
      <CustomParagraph script={script} service={service}>
        {message}
      </CustomParagraph>
      <ul>
        {solutions.map(text => (
          <CustomParagraph
            script={script}
            service={service}
            as="li"
            key={idSanitiser(text)}
          >
            {text}
          </CustomParagraph>
        ))}
      </ul>
      <CustomParagraph script={script} service={service}>
        {callToActionFirst}
        <InlineLink
          href={callToActionLinkUrl}
          className="focusIndicatorReducedWidth"
        >
          {callToActionLinkText}
        </InlineLink>
        {callToActionLast}
      </CustomParagraph>
    </Grid>
  </StyledGelPageGrid>
);

ErrorMain.propTypes = {
  statusCode: string.isRequired,
  title: string.isRequired,
  message: string.isRequired,
  solutions: arrayOf(string).isRequired,
  callToActionFirst: string,
  callToActionLinkText: string.isRequired,
  callToActionLinkUrl: string.isRequired,
  callToActionLast: string,
  script: shape(scriptPropType).isRequired,
  service: string.isRequired,
};

ErrorMain.defaultProps = {
  callToActionFirst: null,
  callToActionLast: null,
};

export default ErrorMain;
