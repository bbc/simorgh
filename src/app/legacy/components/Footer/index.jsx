import React from 'react';
import styled from '@emotion/styled';
import { arrayOf, shape, string, node, bool } from 'prop-types';
import {
  getSansBold,
  getSansRegular,
} from '#psammead/psammead-styles/src/font-styles';
import { getBrevier } from '#psammead/gel-foundations/src/typography';
import {
  GEL_SPACING,
  GEL_SPACING_DBL,
  GEL_MARGIN_BELOW_400PX,
  GEL_MARGIN_ABOVE_400PX,
} from '#psammead/gel-foundations/src/spacings';
import {
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_4_SCREEN_WIDTH_MIN,
} from '#psammead/gel-foundations/src/breakpoints';

import { AmpCookieSettingsButton } from '#containers/ConsentBanner/Banner/cookie.amp';
import Link from './Link';
import List from './List';

const SitewideLinksWrapper = styled.div`
  ${({ script }) => script && getBrevier(script)}
  ${({ service }) => service && getSansRegular(service)}
  background-color: ${props => props.theme.palette.EBON};

  @media (max-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    padding: 0 ${GEL_MARGIN_BELOW_400PX};
  }
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    padding: 0 ${GEL_MARGIN_ABOVE_400PX};
  }
`;

const ConstrainedWrapper = styled.div`
  max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN};
  margin: 0 auto;
  ${({ trustProjectLink }) =>
    trustProjectLink && `padding-top: ${GEL_SPACING};`}
`;

const StyledParagraph = styled.p`
  color: ${props => props.theme.palette.WHITE};
  margin: 0;
  padding: ${GEL_SPACING_DBL} 0;

  // removes padding which creates touch target from the final "inline" link so the Focus Indicator doesn't obscure other text.
  a {
    padding: 0;
  }
`;

const StyledAmpCookieSettingsButton = styled(AmpCookieSettingsButton)`
  ${({ service }) => service && getSansBold(service)}
  background: none;
  border: none;
  color: ${props => props.theme.palette.WHITE};
  cursor: pointer;
  display: block;
  padding: ${GEL_SPACING} 0 ${GEL_SPACING};
  text-decoration: none;
  text-align: left;
  width: 100%;

  &:hover,
  &:focus {
    text-decoration: underline;
  }
`;

const SitewideLinks = ({
  links,
  trustProjectLink,
  copyrightText,
  externalLink,
  isAmp,
  script,
  service,
}) => {
  const elements = links.map(({ id, text, href, lang }) => {
    if (isAmp && id === 'COOKIE_SETTINGS') {
      return (
        <StyledAmpCookieSettingsButton lang={lang} service={service}>
          {text}
        </StyledAmpCookieSettingsButton>
      );
    }
    return <Link service={service} text={text} href={href} lang={lang} />;
  });

  return (
    <SitewideLinksWrapper script={script} service={service}>
      <ConstrainedWrapper trustProjectLink={trustProjectLink}>
        <List
          service={service}
          elements={elements}
          trustProjectLink={trustProjectLink}
        />
        <StyledParagraph>
          {copyrightText}{' '}
          <Link
            service={service}
            text={externalLink.text}
            href={externalLink.href}
            inline
          />
        </StyledParagraph>
      </ConstrainedWrapper>
    </SitewideLinksWrapper>
  );
};

const linkPropTypes = shape({
  href: string.isRequired,
  text: string.isRequired,
});

SitewideLinks.propTypes = {
  links: arrayOf(linkPropTypes.isRequired).isRequired,
  copyrightText: node.isRequired,
  trustProjectLink: linkPropTypes,
  externalLink: linkPropTypes.isRequired,
  isAmp: bool,
  script: shape({}),
  service: string,
};

SitewideLinks.defaultProps = {
  script: null,
  service: null,
  isAmp: false,
  trustProjectLink: null,
};

export default SitewideLinks;
