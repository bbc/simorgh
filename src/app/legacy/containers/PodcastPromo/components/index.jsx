import React from 'react';
import styled from '@emotion/styled';
import { string, shape, arrayOf, element } from 'prop-types';
import { scriptPropType } from '#psammead/gel-foundations/src/prop-types';
import { GEL_SPACING_DBL } from '#psammead/gel-foundations/src/spacings';

import Title from './title';
import Card from './card';
import CardContent from './card-content';
import CardLink from './card-link';
import CardImageWrapper from './card-image-wrapper';
import CardTitle from './card-title';
import CardDescription from './card-description';
import CardEpisodesText from './card-episodes-text';

const PodcastContext = React.createContext({});
const withPodcastContext = Component => props =>
  (
    <PodcastContext.Consumer>
      {context => <Component {...context} {...props} />}
    </PodcastContext.Consumer>
  );

const Wrapper = styled.section`
  background-color: ${props => props.theme.palette.LUNAR};
  padding: ${GEL_SPACING_DBL};
`;

const PodcastPromo = ({ script, service, children, ...props }) => (
  <PodcastContext.Provider value={{ script, service }}>
    <Wrapper {...props}>{children}</Wrapper>
  </PodcastContext.Provider>
);

PodcastPromo.Title = withPodcastContext(Title);
PodcastPromo.Card = Card;
PodcastPromo.Card.Link = CardLink;
PodcastPromo.Card.ImageWrapper = CardImageWrapper;
PodcastPromo.Card.Content = CardContent;
PodcastPromo.Card.Title = withPodcastContext(CardTitle);
PodcastPromo.Card.Description = withPodcastContext(CardDescription);
PodcastPromo.Card.EpisodesText = withPodcastContext(CardEpisodesText);

PodcastPromo.propTypes = {
  children: arrayOf(element).isRequired,
  script: shape(scriptPropType).isRequired,
  service: string.isRequired,
};

PodcastPromo.defaultProps = {};

export default PodcastPromo;
