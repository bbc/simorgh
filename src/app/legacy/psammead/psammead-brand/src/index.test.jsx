import React, { useRef, useEffect } from 'react';
import ScriptLink from '#psammead/psammead-script-link/src';
import { POSTBOX, WHITE } from '../../../../components/ThemeProvider/palette';
import { render } from '../../../../components/react-testing-library-with-providers';
import latin from '../../../../components/ThemeProvider/fontScripts/latin';
import Brand from '.';
import SkipLink from './SkipLink';

const svg = {
  group: (
    <g fillRule="evenodd">
      <path d="M84.32" />
    </g>
  ),
  viewbox: {
    height: 24,
    width: 167.95,
  },
  ratio: 6.9979,
};

describe('Brand', () => {
  it('should render correctly with link provided', () => {
    const { container } = render(
      <Brand
        product="Default Brand Name"
        serviceLocalisedName="Service"
        svgHeight={24}
        maxWidth={280}
        minWidth={180}
        svg={svg}
        url="https://www.bbc.co.uk/news"
        backgroundColour={POSTBOX}
        logoColour={WHITE}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with link not provided', () => {
    const { container } = render(
      <Brand
        product="Default Brand Name"
        serviceLocalisedName="Service"
        svg={svg}
        svgHeight={24}
        maxWidth={280}
        minWidth={180}
        backgroundColour={POSTBOX}
        logoColour={WHITE}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with no service Localised Name', () => {
    const { container } = render(
      <Brand
        product="BBC News"
        svg={svg}
        svgHeight={24}
        maxWidth={280}
        minWidth={180}
        backgroundColour={POSTBOX}
        logoColour={WHITE}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with transparent borders', () => {
    const { container } = render(
      <Brand
        product="BBC News"
        svg={svg}
        svgHeight={24}
        maxWidth={280}
        minWidth={180}
        borderTop
        borderBottom
        backgroundColour={POSTBOX}
        logoColour={WHITE}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  describe('assertions - visually hidden text', () => {
    it('should have role of text when serviceLocalisedName is provided', () => {
      const { container } = render(
        <Brand
          product="Default Brand Name"
          serviceLocalisedName="Service"
          svgHeight={24}
          maxWidth={280}
          minWidth={180}
          svg={svg}
          url="https://www.bbc.co.uk/news"
          backgroundColour={POSTBOX}
          logoColour={WHITE}
        />,
      );

      expect(container.querySelector('span').getAttribute('role')).toEqual(
        'text',
      );
    });

    it('should not have role of text when serviceLocalisedName is not provided', () => {
      const { container } = render(
        <Brand
          product="Default Brand Name"
          svgHeight={24}
          maxWidth={280}
          minWidth={180}
          svg={svg}
          url="https://www.bbc.co.uk/news"
          backgroundColour={POSTBOX}
          logoColour={WHITE}
        />,
      );

      expect(container.querySelector('span').getAttribute('role')).toBeNull();
    });

    it('should add extra props passed to the component', () => {
      const { container } = render(
        <Brand
          product="Default Brand Name"
          svgHeight={24}
          maxWidth={280}
          minWidth={180}
          svg={svg}
          url="https://www.bbc.co.uk/news"
          backgroundColour={POSTBOX}
          logoColour={WHITE}
          data-brand="header"
        />,
      );

      expect(container.querySelector('div').getAttribute('data-brand')).toEqual(
        'header',
      );
    });

    it('should let the brand link be focussed with a ref', () => {
      const TestComponent = () => {
        const brandRef = useRef(null);

        useEffect(() => {
          brandRef.current?.querySelector('a')?.focus();
        }, []);

        return (
          <Brand
            product="Default Brand Name"
            svgHeight={24}
            maxWidth={280}
            minWidth={180}
            svg={svg}
            url="https://www.bbc.co.uk/news"
            backgroundColour={POSTBOX}
            logoColour={WHITE}
            data-brand="header"
            ref={brandRef}
          />
        );
      };
      const initialFocus = document.activeElement;
      const { container } = render(<TestComponent />);
      const brand = container.querySelector('a');

      expect(document.activeElement).toBe(brand);
      expect(document.activeElement).not.toBe(initialFocus);
    });

    it('should render brand link with id where linkId and url props are provided', () => {
      const { container } = render(
        <Brand
          product="BBC News"
          svg={svg}
          url="https://www.bbc.co.uk/news"
          svgHeight={24}
          maxWidth={280}
          minWidth={180}
          borderTop
          borderBottom
          backgroundColour={POSTBOX}
          logoColour={WHITE}
          linkId="brandLink"
        />,
      );

      const brandLink = container.querySelector('#brandLink');
      expect(brandLink).toBe(container.querySelector('a'));
    });

    it('should render script, frontpage and skip to content links', () => {
      const scriptLinkComponent = (
        <ScriptLink
          script={latin}
          service="serbian"
          href="https://www.bbc.com/serbian/lat"
        >
          Lat
        </ScriptLink>
      );

      const skipLink = (
        <SkipLink service="news" script={latin} href="#content">
          Skip to content
        </SkipLink>
      );

      const { container } = render(
        <Brand
          product="Default Brand Name"
          svgHeight={24}
          maxWidth={280}
          minWidth={180}
          svg={svg}
          url="https://www.bbc.co.uk/news"
          backgroundColour={POSTBOX}
          logoColour={WHITE}
          skipLink={skipLink}
          data-brand="header"
          scriptLink={scriptLinkComponent}
        />,
      );

      const links = container.querySelectorAll('a');
      expect(links).toHaveLength(3);

      const frontpageLink = links[0];
      expect(frontpageLink.getAttribute('href')).toEqual(
        'https://www.bbc.co.uk/news',
      );
      expect(frontpageLink.textContent).toEqual('Default Brand Name');

      const skipToContentLink = links[1];
      expect(skipToContentLink.getAttribute('href')).toEqual('#content');
      expect(skipToContentLink.textContent).toEqual('Skip to content');

      const scriptLink = links[2];
      expect(scriptLink.getAttribute('href')).toEqual(
        'https://www.bbc.com/serbian/lat',
      );
      expect(scriptLink.textContent).toEqual('Lat');
    });
  });
});
