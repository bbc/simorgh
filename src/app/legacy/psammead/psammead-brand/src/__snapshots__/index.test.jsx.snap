// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand should render correctly with link not provided 1`] = `
.emotion-0 {
  background-color: #B80000;
  height: 2.75rem;
  width: 100%;
  padding: 0 0.5rem;
}

@media (min-width: 25rem) {
  .emotion-0 {
    height: 3.5rem;
    padding: 0 1rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-0 {
    height: 4rem;
  }
}

.emotion-2 {
  height: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-width: 63rem;
  margin: 0 auto;
}

@media (max-width: 14.9375rem) {
  .emotion-2 {
    display: block;
  }
}

.emotion-4 {
  box-sizing: content-box;
  color: #FFFFFF;
  fill: currentColor;
  height: 1.25rem;
}

@media (min-width: 25rem) {
  .emotion-4 {
    height: 1.5rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-4 {
    height: 1.875rem;
  }
}

@media screen and (-ms-high-contrast: active),print {
  .emotion-4 {
    fill: windowText;
  }
}

.emotion-6 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <svg
        aria-hidden="true"
        class="emotion-4 emotion-5"
        focusable="false"
        height="32"
        viewBox="0 0 167.95 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill-rule="evenodd"
        >
          <path
            d="M84.32"
          />
        </g>
      </svg>
      <span
        class="emotion-6 emotion-7"
        id="BrandLink-null"
        role="text"
      >
        <span
          lang="en-GB"
        >
          Default Brand Name, 
        </span>
        <span>
          Service
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`Brand should render correctly with link provided 1`] = `
.emotion-0 {
  background-color: #B80000;
  height: 2.75rem;
  width: 100%;
  padding: 0 0.5rem;
}

@media (min-width: 25rem) {
  .emotion-0 {
    height: 3.5rem;
    padding: 0 1rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-0 {
    height: 4rem;
  }
}

.emotion-2 {
  height: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-width: 63rem;
  margin: 0 auto;
}

@media (max-width: 14.9375rem) {
  .emotion-2 {
    display: block;
  }
}

.emotion-4 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  bottom: 0.125rem;
  padding-top: 0.125rem;
}

.emotion-4:hover,
.emotion-4:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 0.25rem solid #FFFFFF;
  margin-bottom: -0.25rem;
}

.emotion-4:focus-visible::after {
  content: '';
  position: absolute;
  top: 0;
  left: -0.1875rem;
  bottom: 0;
  right: -0.1875rem;
  border-top: 0.25rem solid #FFFFFF;
  outline: 0.25rem solid #FFFFFF;
}

.emotion-6 {
  box-sizing: content-box;
  color: #FFFFFF;
  fill: currentColor;
  height: 1.25rem;
}

@media (min-width: 25rem) {
  .emotion-6 {
    height: 1.5rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-6 {
    height: 1.875rem;
  }
}

@media screen and (-ms-high-contrast: active),print {
  .emotion-6 {
    fill: windowText;
  }
}

.emotion-8 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <a
        aria-labelledby="BrandLink-null"
        class="focusIndicatorRemove emotion-4 emotion-5"
        href="https://www.bbc.co.uk/news"
      >
        <svg
          aria-hidden="true"
          class="emotion-6 emotion-7"
          focusable="false"
          height="32"
          viewBox="0 0 167.95 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill-rule="evenodd"
          >
            <path
              d="M84.32"
            />
          </g>
        </svg>
        <span
          class="emotion-8 emotion-9"
          id="BrandLink-null"
          role="text"
        >
          <span
            lang="en-GB"
          >
            Default Brand Name, 
          </span>
          <span>
            Service
          </span>
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`Brand should render correctly with no service Localised Name 1`] = `
.emotion-0 {
  background-color: #B80000;
  height: 2.75rem;
  width: 100%;
  padding: 0 0.5rem;
}

@media (min-width: 25rem) {
  .emotion-0 {
    height: 3.5rem;
    padding: 0 1rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-0 {
    height: 4rem;
  }
}

.emotion-2 {
  height: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-width: 63rem;
  margin: 0 auto;
}

@media (max-width: 14.9375rem) {
  .emotion-2 {
    display: block;
  }
}

.emotion-4 {
  box-sizing: content-box;
  color: #FFFFFF;
  fill: currentColor;
  height: 1.25rem;
}

@media (min-width: 25rem) {
  .emotion-4 {
    height: 1.5rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-4 {
    height: 1.875rem;
  }
}

@media screen and (-ms-high-contrast: active),print {
  .emotion-4 {
    fill: windowText;
  }
}

.emotion-6 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <svg
        aria-hidden="true"
        class="emotion-4 emotion-5"
        focusable="false"
        height="32"
        viewBox="0 0 167.95 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill-rule="evenodd"
        >
          <path
            d="M84.32"
          />
        </g>
      </svg>
      <span
        class="emotion-6 emotion-7"
        id="BrandLink-null"
      >
        BBC News
      </span>
    </div>
  </div>
</div>
`;

exports[`Brand should render correctly with transparent borders 1`] = `
.emotion-0 {
  background-color: #B80000;
  height: 2.75rem;
  width: 100%;
  padding: 0 0.5rem;
  border-top: 0.0625rem solid transparent;
  border-bottom: 0.0625rem solid transparent;
}

@media (min-width: 25rem) {
  .emotion-0 {
    height: 3.5rem;
    padding: 0 1rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-0 {
    height: 4rem;
  }
}

.emotion-2 {
  height: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  max-width: 63rem;
  margin: 0 auto;
}

@media (max-width: 14.9375rem) {
  .emotion-2 {
    display: block;
  }
}

.emotion-4 {
  box-sizing: content-box;
  color: #FFFFFF;
  fill: currentColor;
  height: 1.25rem;
}

@media (min-width: 25rem) {
  .emotion-4 {
    height: 1.5rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-4 {
    height: 1.875rem;
  }
}

@media screen and (-ms-high-contrast: active),print {
  .emotion-4 {
    fill: windowText;
  }
}

.emotion-6 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <svg
        aria-hidden="true"
        class="emotion-4 emotion-5"
        focusable="false"
        height="32"
        viewBox="0 0 167.95 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill-rule="evenodd"
        >
          <path
            d="M84.32"
          />
        </g>
      </svg>
      <span
        class="emotion-6 emotion-7"
        id="BrandLink-null"
      >
        BBC News
      </span>
    </div>
  </div>
</div>
`;
